src/msg/async/AsyncConnection.cc

```diff
@@ -821,6 +816,8 @@ void AsyncConnection::process()
     }
   } while (prev_state != state);
 
+  if (need_dispatch_writer)
```

Comment:
no write_lock needed. can_write is a atomic variable and read access is allowed without lock. only modify is lock needed